firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -107,12 +107,21 @@
   }
 
   /**
-   * Perform pre-condition checks to make sure {@link FirebaseOptions#getApiKey()} and {@link
-   * FirebaseOptions#getProjectId()} are not null.
+   * Perform pre-condition checks to make sure {@link FirebaseOptions#getApiKey()} and ({@link
+   * FirebaseOptions#getProjectId()} or {@link FirebaseOptions#getGcmSenderId()} ()} are not null.
    */
   private void preConditionChecks() {
+    Preconditions.checkNotNull(getApplicationId());
```

Comment:
Did you define the method #getApplicationId?

Reply:
It was defined previously. git only shows the changes - you can see the method: https://github.com/firebase/firebase-android-sdk/blob/cbb626b2776de344076d37b5b4fb03d6643c30bc/firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java#L143